### DISCUSSION TITLE
accountability in schedule hook

### DISCUSSION THREAD

**Discussion by 3zzy** [directus/directus#15511] [Archive]
`filter` and `action` hook context has `accountability` object available however `schedule` doesn't. 

I need to create items within the schedule hook and while the following works, but the `user_created` is `null`:

```
export default (
  { filter, action, init, schedule },
  { services, exceptions, database, getSchema, env, logger, emitter }
) => {

  const { InvalidPayloadException, ServiceUnavailableException, } = exceptions;
  const { UsersService, ItemsService } = services;

  schedule('*/1 * * * *', async () => {

    const schema = await getSchema();
    const invoices = new ItemsService("invoices", {
      database,
      schema,
      // accountability ???
    });

  });
}
```

I tried creating a `UserService` and passing that but doesn't work either. 

Any help would be appreciated. Thanks!

**Comment by azrikahar** [comment]
Note: I've updated the `subscribe` word in the title and in your post to `schedule`, since I believe that's what you're referring to based on your code & context.

---

It is working as expected to _not_ have `accountability` for `schedule` hooks because it's executed by Directus itself (or system if that makes sense), and there's no "user" in this scenario, which is why it _should_ be null. 

Is there a particular issue/expectation that prompts the need to have an accountability? Perhaps revisions? On a related note, and assuming revision is the reason why you wish to have some form of accountability indicate it as a "system" user, #15467 should somewhat indirectly support this particular use case going forward.

**Reply by 3zzy** [reply]
Yes I meant `schedule`, thanks for correcting that.

Exactly. I need the 'accountability' to indicate it was a "system" user which I guess could be inferred from an empty field as well however I have multiple extensions doing various tasks so I need to know exactly which "system" user was it. 

Looks like that PR is almost ready so thats great news! Thanks!
